Frontend api list return withdraw

deerman31 opened the pull request:
フロントエンドの以下の部分はまだ未完成。
1. cssの修正
2. トランザクションが起きた後のリストの再表示。
3. スマートコントラクトをDevnetに変更した後に繋げられるようにフロントエンドのコードを書き換える。
4. add new transactionのバリデーション

上記以外は完成しているので一旦mainにマージします。

deerman31 committed:
校舎のpcでテストするためにpush

deerman31 committed:
ReturnSolButtonに引数を渡す。

deerman31 committed:
一次的に

deerman31 committed:
校舎pcで実行

deerman31 committed:
校舎でテスト

deerman31 committed:
linterを無視

deerman31 committed:
requireへの警告を無視するようにした。

deerman31 committed:
校舎でテスト

deerman31 committed:
test

deerman31 committed:
校舎でテスト

deerman31 committed:
rikedaさんのローカルでテスト

deerman31 committed:
build時間の短縮のためにDockerfileを修正

deerman31 committed:
api.tsの修正

deerman31 committed:
再度

deerman31 committed:
/Users/yoshinarikusano/Developments/Jin/srcs/app/frontend/src/app/Providers/WalletProvider.tsx
でlocalnetに変更。

deerman31 committed:
a

Commit:
挙動OK

rikeda-cloud committed:
[update] replaceでList表示するtsxファイルも置換対象に変更

rikeda-cloud committed:
[update] List表示可能

rikeda-cloud committed:
[update] format && AddNewTransaction実行時に、自動的にトランザクションの数+1をtransactionIdとして生成

deerman31 committed:
テストするためにpush

deerman31 committed:
withdrawボタンをテスト

rikeda-cloud committed:
[fix] withdrawされたかの判定条件を修正

deerman31 committed:
buyerとsellerでaddNewTransactionが表示できるようにした

deerman31 committed:
必要ないimportを削除

tachu2 commented on srcs/app/debug_tool/entrypoint.sh:
```diff
@@ -21,8 +21,8 @@ main() {
   elif [ "${cmd}" = "lint" ]; then
-    echo "Linting backend..."
-    cargo clippy --manifest-path ${TOML_FILE} || return 1
+    # echo "Linting backend..."
+    # cargo clippy --manifest-path ${TOML_FILE} || return 1
```
これ一時的にコメントアウトしたやつですか？
今はlintterは動かしても問題無いですか？

deerman31 replied:
rustでコンパイルされている時点で問題ないと認識しているため、linter(clippy)はしない方向性に変えました。

tachu2 replied:
lintterはcoding ruleを統一するためのものなので、lintterは実行すべきだと思います。（そのために初めに環境構築しました）
のちのrefactorの対応でも大丈夫です。

rikeda-cloud replied:
Rustではコンパイル時にエラーがわかるのでコンパイルによってCodingRuleは統一されていると思います。
clippyというLinterツールはより細かいエラーをはくので、どちらに合わせるかはありますが、私は開発時にRustのコンパイルが通っていることを基準にコーディングしていました。
clippyに合わせる必要はないと思っています

tachu2 commented on srcs/app/frontend/.eslintrc.json:
```diff
@@ -6,8 +6,12 @@
-    "prettier/prettier": "error",
-    "@typescript-eslint/no-unused-vars": "error",
-    "@typescript-eslint/no-explicit-any": "warn"
+    //"prettier/prettier": "error",
+    "prettier/prettier": "off",
+    //"@typescript-eslint/no-unused-vars": ["error", { "argsIgnorePattern": "^_" }],
+    "@typescript-eslint/no-unused-vars": "off",
+    "@typescript-eslint/no-var-requires": "off",
+    //"@typescript-eslint/no-explicit-any": "warn"
+    "@typescript-eslint/no-explicit-any": "off"
```
これも一時的に変えているだけですか？

deerman31 replied:
一時的に変えているだけです。
修正しました。

tachu2 commented on srcs/app/frontend/sample/test.js:
```diff
@@ -20,174 +27,229 @@ const SECONDS = 10;
-		is_canceled: isCanceled,
-		user_defined_data: userDefinedData,
-	};
+  const sellerPubkey = buffer.slice(8, 40);
+  const buyerPubkey = buffer.slice(40, 72);
+  const transactionId = buffer.readBigUInt64LE(72);
+  const amountLamports = buffer.readBigUInt64LE(80);
+  const createAt = buffer.readBigInt64LE(88);
+  const refundDeadline = buffer.readBigInt64LE(96);
+  const isCanceled = buffer.readUInt8(104) !== 0;
+  const userDefinedData = buffer.slice(109).toString('utf-8');
+
```
ここらへんマジックナンバーだらけなので、コメントがほしいです。

tachu2 replied:
これはtestファイルでした。
無視してください。

tachu2 commented on srcs/app/frontend/src/app/components/Home/Body/Buyer/ClaimsList.tsx:
```diff
@@ -135,3 +98,71 @@ const ClaimsList = () => {
+}
+
+function decodeRefundableEscrow(buffer: Buffer): RefundableEscrowData {
+  const sellerPubkey = buffer.slice(8, 40);
+  const buyerPubkey = buffer.slice(40, 72);
+  const transactionId = buffer.readBigUInt64LE(72);
+  const amountLamports = buffer.readBigUInt64LE(80);
+  const createAt = buffer.readBigInt64LE(88);
+  const refundDeadline = buffer.readBigInt64LE(96);
+  const isCanceled = buffer.readUInt8(104) !== 0;
+  const userDefinedData = buffer
+    .slice(109)
+    .toString('utf-8')
+    .replace(/\u0000/g, '')
+    .trim();
+
```
ここのマジックナンバーに対して、コメントがほしいです。

rikeda-cloud replied:
了解です

tachu2 commented on srcs/app/frontend/src/app/components/Home/Body/Buyer/UnrecoverableList.tsx:
```diff
@@ -7,95 +7,106 @@ srcs/app/frontend/src/app/components/Body/Buyer/UnrecoverableList.tsx
+}
+
+function decodeRefundableEscrow(buffer: Buffer): NotRetTransaction {
+  const sellerPubkey = buffer.slice(8, 40);
+  const transactionId = buffer.readBigUInt64LE(72);
+  const amountLamports = buffer.readBigUInt64LE(80);
+  const userDefinedData = buffer
+    .slice(109)
+    .toString('utf-8')
+    .replace(/\u0000/g, '')
+    .trim();
+  const refundDeadline = buffer.readBigInt64LE(96);
```
これも同様に、マジックナンバーです。
共通の定数でまとめられそう？

rikeda-cloud replied:
Seller側ではbuyerPubkeyを取得し、
Buyer側ではsellerPubkeyを取得するので、関数は共通化できないです。
現状、Seller側でもBuyer側でもsellerPubkeyという名前にしているのでそこは修正します。

rikeda-cloud replied:
enum型でマジックナンバーを定義し、importする形でマジックナンバーを定数に置き換えました。

tachu2 reviewed (commented):
対応ありがとうございます。
気になった点コメントしました。

deerman31 committed:
cssを修正

deerman31 committed:
eslintrc.jsonでlinterを抑制していたが、提出のため、抑制をoff

deerman31 committed:
make fmtを行った。

rikeda-cloud committed:
[update] マジックナンバーをenum定数に変更。ビルドできるように修正

rikeda-cloud committed:
[fix] マジックナンバーの修正間違いを修正

tachu2 commented on srcs/app/frontend/src/app/Providers/WalletProvider.tsx:
```diff
@@ -5,26 +5,32 @@ import {
-  const endpoint = useMemo(() => clusterApiUrl(network), [network]);
+  // Localnetのエンドポイントを指定
+  const endpoint = 'http://localhost:8899';
+  //const network = WalletAdapterNetwork.Devnet;
+  //const endpoint = useMemo(() => clusterApiUrl(network), [network]);
```
不要なコード？（コメントアウト）が残っています。

deerman31 replied:
後々、devnetに切り替える際の情報として残しています。

現在,localnetで行っていますが、後でdevnetにも切り替える必要がありますので。

tachu2 replied:
必要なのであれば、そのような意図をコメントで残すのがいいと思います。

tachu2 commented on srcs/app/frontend/src/app/components/Home/Body/Buyer/ClaimsList.tsx:
```diff
@@ -8,94 +8,51 @@ srcs/app/frontend/src/app/components/Body/Buyer/ClaimsList.tsx
-  const [transactions, setTransactions] = useState<ReturnableTransaction[]>([]);
+  const [transactions, setTransactions] = useState<RefundableEscrowData[]>([]);
+  const { publicKey } = useWallet();
+  //const nowDate = getCurrentDate();
 
```
不要なら削除でいいと思います。

deerman31 replied:
削除します。

deerman31 committed:
テストするためにpush

deerman31 committed:
make lint, fmtを行った後にイメージとcontainerを削除しないと
後々、make fmtで古いコンテナからローカルにファイルをコピーしてしまう恐れがあるため、make lintとmake fmtの後にmake downを行うようにした。

weijuan82113 commented:
一度目を通りました、動作を動けることを優先にしてください。

weijuan82113 commented on srcs/app/build_tool/replace/src/main.rs:
```diff
@@ -18,9 +18,9 @@ fn replace_content(program_id: &str, replace_path: &str) -> io::Result<()> {
-            "const PROGRAM_ID = new PublicKey('Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS');",
-            &format!("const PROGRAM_ID = new PublicKey('{}');", program_id),
+        Some("js") | Some("ts") | Some("tsx") => replace_content.replacen(
+            "Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS",
```
なぜかPROMGRAM_IDがハードコーディングになったですか？

weijuan82113 commented on srcs/app/frontend/sample/test.js:
```diff
@@ -20,174 +27,229 @@ const SECONDS = 10;
+    transactionIdBuffer,
+  ];
+
+  const [counterPDA, _] = await PublicKey.findProgramAddress(seeds, PROGRAM_ID);
```
下記の記述方法の方がわかりやすいです
const [counterPDA, _] = await PublicKey.findProgramAddress(
seeds,
PROGRAM_ID
);
